### DISCUSSION TITLE
help appreciated

### DISCUSSION THREAD

**Discussion by vGelder** [patrickvorgers/Home-Assistant-Import-Energy-Data#28] [Ideas]
Hi Patrick,

been trying to import  some daily data form the past 6 months, as my data is not from Toon but from Oxxio I created the .CSV files using excel. With these i followed your instructions. I can see that the *_NEW tables are created and filled but somehow the data is not imported into the statistics and statistics_short_term tables. even as executing the SQL  script doesn't report any errors.

As I'm not a developer and lack the skills, I'm hoping you can help me out and put me in the right direction. 
attached: the .CSV files, database and SQLlite project.

thank you!

[files.zip](https://github.com/patrickvorgers/HA-Import-Toon-Data/files/13833888/files.zip)

**Comment by patrickvorgers** [comment]
Hi,

I looked at the script. The daily data from the CSV files was not imported into the *_NEW tables because the import of the hourly data was commented out. It first checks if it already has information with a higher resolution before importing the daily information. I changed the script and removed the check (see attached script).

The CSV files do not contain a lot of data is that correct?

[Script.zip](https://github.com/patrickvorgers/HA-Import-Toon-Data/files/13834740/Script.zip)

**Comment by vGelder** [comment]
thanks for your swift reply!

my CSV files are indeed quite short, the solar panels are only two months old and the current provider is since last June. Once I got this data in, I look into the older data.

I will check the new script this weekend, keep you posted.

**Comment by vGelder** [comment]
Hi,

I got it almost right.., all except the solar data. I'm a bit confused here, what is expected in the .CSV? is it daily data in Wh like this:

1699142400,0
1699228800,743
1699315200,20070
1699401600,8936
1699488000,15690
1699574400,3098
1699660800,14162
1699747200,4814
1699833600,2420
1699920000,5364
1700006400,13160
1700092800,1594
1700179200,20897
1700265600,931

or cumulative daily data in Wh like this:

1699142400,0
1699228800,743
1699315200,20813
1699401600,29749
1699488000,45439
1699574400,48537
1699660800,62699
1699747200,67513
1699833600,69933
1699920000,75297
1700006400,88457
1700092800,90051
1700179200,110948
1700265600,111879

both give some unexpected results?

thanks!

**Comment by vGelder** [comment]
Oke... I got it!

the correct solar data was cumulative and in wH * 1000 like beneath. this resulted in a script error because in the "cleanup wrongs" part there was a control check: - Diff > 1000	=> Incorrect value.

eventually I got all the historical data restored.
tnx!

elec_solar_quantity_CurrentElectricityQuantity_10yrdays.csv:
1699142400,0
1699228800,743000
1699315200,20813000
1699401600,29749000
1699488000,45439000
1699574400,48537000
1699660800,62699000
1699747200,67513000
1699833600,69933000
1699920000,75297000
1700006400,88457000
1700092800,90051000
1700179200,110948000
1700265600,111879000

**Comment by vGelder** [comment]
footnote:

1) Maybe its possible to update the readme and add the importance to make a backup before tweaking the database ;) I totaly forgot........
2) in the script there are several places where I needed to /*change /* the sensor ID's, is it possible to make use of variables? because i forgot to change one or two, and it screwed up the database........

anyway Tnx-A-Lot!

**Comment by patrickvorgers** [comment]
Last weekend I completly rewrote the script and incorporated your feedback. I made the script generic and more robust.

Can you share with me how you prepared your data from Oxxio so that I can add is as a datasource (see new setup)?

**Comment by vGelder** [comment]
Hi Patrick,

I'm looking into your request. Been preparing a new dataset based on Oxxio and the new script. Do to a lack of time i hope to have some results next week.

Regards!

**Comment by patrickvorgers** [comment]
@vGelder Thank you very much!
It would already help just for starters to have a sample file.

**Comment by vGelder** [comment]
Hi,

I got the following files for you:

At 'mijn-0xxio.nl' I got two options. the first is to download the data direct from oxxio which gave me the following XLS:

[Oxxio Verbruik_05-06-2023-22-01-2024.xlsx](https://github.com/patrickvorgers/Home-Assistant-Import-Energy-Data/files/14014456/Oxxio.Verbruik_05-06-2023-22-01-2024.xlsx)

the second option was to request the data from the 'netwerkbeheerder' in my case that is Liander they gave the following CSV:

[csv_bijlage.csv](https://github.com/patrickvorgers/Home-Assistant-Import-Energy-Data/files/14014555/csv_bijlage.csv)

each file represents only the time I'm an Oxxio customer.


Prior to Oxxio I was a customer of Engie, from their App I downloaded the following CSV:

[Export verbruiken ENGIE.csv](https://github.com/patrickvorgers/Home-Assistant-Import-Energy-Data/files/14014585/Export.verbruiken.ENGIE.csv)

note that during the Engie period I didn't produce energy jet so that data is not present.


Using the URL provided by @ndewijer: (https://github.com/ndewijer/HA-Import-Historical-Data/tree/main/SolarEdge_API) I got the following output for solar data:

[Solar Edge.txt](https://github.com/patrickvorgers/Home-Assistant-Import-Energy-Data/files/14014772/Solar.Edge.txt)

Hope this helps you with the script.

Please feel free if I can help you with some testing.

**Comment by vGelder** [comment]
New request ;)

I have now the data of multiple energy providers imported into HA which is great.  Maybe it's also possible to add the costs according the imported data?

I was thinking that it shout be possible to 

1) add a table with the columns change_date and new_costs
2) delete currents costs from database
3) insert new costs into the database

Obviously I not a developer... ;) but what do you think?

**Comment by vGelder** [comment]
New request ;)

I have now the data of multiple energy providers imported into HA which is great.  Maybe it's also possible to add the costs according the imported data?

I was thinking that it shout be possible to 

1) add a table with the columns change_date and new_costs
2) delete currents costs from database
3) insert new costs into the database

Obviously I'm not a developer...:) but what do you think?

**Comment by patrickvorgers** [comment]
Interesting idea, I will have to dive into it to determine whether it is doable.

Thank you for the sample data. I already created the script for Engie and Oxxio and working on Liander.
